apps/gradient_autoscheduler/GradientAutoscheduler.cpp

```diff
@@ -154,6 +154,59 @@ void parallelize_vars_and_rvars_gpu(
         serial_rvars = rvars;
     }
 
+    if (gpu_threads.empty() && r_gpu_threads.empty()) {
+        // If we can't find any loop large enough for a GPU thread,
+        // use the largest loop as the GPU thread.
+        int loop_size = 0;
+        int largest_loop_id = -1;
+        for (int i = 0; i < (int)vars.size(); i++) {
+            if (var_bounds[i] > loop_size) {
+                loop_size = var_bounds[i];
+                largest_loop_id = i;
+            }
+        }
+        for (int i = 0; i < (int)rvars.size(); i++) {
+            if (rvar_bounds[i] > loop_size) {
+                loop_size = rvar_bounds[i];
+                largest_loop_id = i + (int)vars.size();
+            }
+        }
+        if (largest_loop_id != -1) {
+            if (largest_loop_id < (int)vars.size()) {
+                // The largest loop is a pure variable
+                const Var &v = vars[largest_loop_id];
+                Var inner;
+                func_or_stage.split(v,
+                                    v,
+                                    inner,
+                                    32,  // warp size
```

Comment:
How is 32 chosen? Does it deserve to be a named constant?

Reply:
Named!